docs/binding-controller.md

```diff
@@ -0,0 +1,60 @@
+```mermaid
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+    classDef Red fill:#FF9999;
+    classDef Amber fill:#FFDEAD;
+    classDef Green fill:#BDFFA4;
+
+  predicate((PREDICATE: <br>SnapshotEnvironmentBinding<br>is created or updated))
+
+%%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureIntegrationTestPipelineForScenarioExists() function
+
+%% Node definitions
+ensure1(Proceed further if:<br>Snapshot testing <br> not finished yet)
+isThereAnITS("Is there an<br>IntegrationTestScenario<br>present in the binding?")
+continueProcessing1[/Controller continues processing.../]
+getLatestPipelineRun("Get latest pipeline run for<br>snapshot and scenario")
+isPipelineRunExisting("Does an integration<br>pipeline run exist?")
+createNewPipelineRun("Create a new pipeline run<br>for the snapshot")
+
+
+%% Node connections
+predicate               ----> |"EnsureIntegrationTestPipelineForScenarioExists()"|ensure1
+ensure1               ---->       isThereAnITS
+isThereAnITS            --No-->     continueProcessing1
+isThereAnITS            --Yes-->    getLatestPipelineRun
+getLatestPipelineRun    ---->       isPipelineRunExisting
+isPipelineRunExisting   --No-->     continueProcessing1
+isPipelineRunExisting   --Yes-->    createNewPipelineRun
+createNewPipelineRun    ---->       continueProcessing1
+
+%%%%%%%%%%%%%%%%%%%%%%% Drawing createIntegrationPipelineRunWithEnvironment() function
+
+%% Node definitions
+getTarget(Get deployment target for<br>environment)
+createPipelineRun(Create new pipelineRun for<br>application/snapshot/environment combination)
```

Comment:
What do you think about explicitly mentioning that we do add a special parameter ("NAMESPACE") and workspace binding ("cluster-credentials") which takes the value from DT? That's what makes [this Test PLR](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/binding/snapshotenvironmentbinding_adapter.go#L115) special than the one [created by the snapshot controller](https://github.com/redhat-appstudio/integration-service/blob/7b264cf65de4cc2b5bce4633a6ef415a12fcde36/controllers/snapshot/snapshot_adapter.go#L469). Wdyt?

Reply:
I think that gets too much into implementation details.  In fact, I think I should remove or modify the `application/snapshot/environment` bit.

Reply:
makes sense :+1:

Reply:
It could be good to mention `snapshot/ITS/environment` combination that uniquely identifies the pipelineRun.